samples/ingestion_biru_example.py

```diff
@@ -36,79 +32,181 @@
+        )
+
+    def run_extract(self) -> IngestibleDataclasses:
+        """
+        Run the extraction plant process.
+
+        Returns:
+            IngestibleDataclasses: Result of the extraction plant process.
+        """
+        prospector = Prospector(self.profile_loader.load_custom_prospector())
+        miner = Miner(self.profile_loader.load_custom_miner())
+        beneficiation = Beneficiation(self.profile_loader.load_custom_beneficiation())
+        ext_plant = ExtractionPlant(prospector, miner, beneficiation)
+        ingestibles = ext_plant.extract(self.yaml_path)
+        return ingestibles
+
+    def run_ingestion(self, ingestible_dataclass: IngestibleDataclasses) -> None:
+        """
+        Run the ingestion process.
+
+        Args:
+            ingestible_dataclasses (IngestibleDataclasses): Ingestible data classes
+            containing data to be updated.
+        """
+        new_ingested_datafiles = []
+
+        try:
+            for project in ingestible_dataclass.get_projects():
+                self.update_data_status(project, "project", new_ingested_datafiles)
+
+            for experiment in ingestible_dataclass.get_experiments():
+                self.update_data_status(
+                    experiment, "experiment", new_ingested_datafiles
+                )
+
+            for dataset in ingestible_dataclass.get_datasets():
+                self.update_data_status(dataset, "dataset", new_ingested_datafiles)
+
+            for datafile in ingestible_dataclass.get_datafiles():
+                self.update_data_status(datafile, "datafile", new_ingested_datafiles)
+
+        except TimeoutError as e:
+            logger.error(e)
+
+        # TODO: write the updated ingestible_dataclasses into a YAML file
+        # YAMLUpdater(self.yaml_path.as_posix(), ingestible_dataclass)
+
+        # Initiate Conveyor
+        self.initiate_conveyor(new_ingested_datafiles)
+
+    def update_data_status(
+        self,
+        data_obj: Union[RawProject, RawExperiment, RawDataset, RawDatafile],
+        obj_type: str,
+        datafile_list: list,
+    ) -> None:
+        """
+        Update the data status for a specific object type.
+
+        Args:
+            data_obj (Union[RawProject, RawExperiment, RawDataset, RawDatafile]):
+                The data object to update.
+            obj_type (str): The type of the data object.
+            new_ingested_datafiles (list): List to store newly ingested datafiles.
+        """
+        if (
+            data_obj.data_status is not None
+            and data_obj.data_status.value == DataStatus.INGESTED.value
+        ):
+            obj_name_attr = obj_type.lower()
+            logger.info("%s %s has already been ingested.", obj_type, obj_name_attr)
+            return
+
+        obj_name_attr = obj_type.lower()
+        logger.info("Ingesting %s: %s", obj_type.lower(), obj_name_attr)
+        result = getattr(self.factory, f"ingest_{obj_type}s")([data_obj])
```

Comment:
I found it a bit confusing that the actual ingestion is buried inside a method called `update_data_status()`, and it took me quite a while to figure out that this line is actually invoking a function, so maybe there's room for some refactoring here. Might be easiest to discuss in person

Reply:
Just adjusted a little bit, but might need to refactor anyway.